src/dllmain.cpp

```diff
@@ -697,11 +702,60 @@ void GraphicalTweaks()
                 {
                     if ((ctx.rcx + 0x180) && (ctx.rcx + 0x184))
                     {
-                        // Persona model previews = 1024x1024
-                        if ((*reinterpret_cast<int*>(ctx.rcx + 0x180) == 1024) && (*reinterpret_cast<int*>(ctx.rcx + 0x184) == 1024))
+						int* pRenTexSizeX = reinterpret_cast<int*>(ctx.rcx + 0x180);
+						int* pRenTexSizeY = reinterpret_cast<int*>(ctx.rcx + 0x184);
+						int iRenTexSizeX = *pRenTexSizeX;
+						int iRenTexSizeY = *pRenTexSizeY;
+						float fRenTexRatioX = (float)iRenTexSizeX / (float)iRenTexSizeY;
+						float fRenTexRatioY = (float)iRenTexSizeY / (float)iRenTexSizeX; // also 1 / X
+                        if (   (iRenTexSizeX >= 32 && iRenTexSizeX < iCustomResX)
+							|| (iRenTexSizeY >= 32 && iRenTexSizeY < iCustomResY))
                         {
-                            *reinterpret_cast<int*>(ctx.rcx + 0x180) *= fRenTexResMulti;
-                            *reinterpret_cast<int*>(ctx.rcx + 0x184) *= fRenTexResMulti;
+							int iMaxTexSize = 4096; // can technically go up to 16384, however anything past 4096 causes notable stutters when the render texture is loaded 
+
+							spdlog::info("----------");
+							spdlog::info("Render Texture 2D: X {:d}, Y {:d}, Ratio {:f}", iRenTexSizeX, iRenTexSizeY, fRenTexRatioX);
+
+							if (fRenTexRatioX == fAspectRatio) {
+								spdlog::info("Render Texture 2D has same aspect as screen: {:d}, {:d}", iRenTexSizeX, iRenTexSizeY);
+							}
+
+							if (iRenTexSizeX == 1920 && iRenTexSizeY == 1080) {
+								// Seems to affect pause menu screens, has some minor bugs in transitions
+								float fRenderScale = fScreenPercentage / 100.0f;
+								iRenTexSizeX = iCustomResX * fRenderScale;
+								iRenTexSizeY = iCustomResY * fRenderScale;
```

Comment:
```suggestion
float fRenderScale = fScreenPercentage / 100.0f;
iRenTexSizeX = iCustomResX * fRenderScale;
iRenTexSizeY = iCustomResY * fRenderScale;

if (fAspectRatio > fNativeAspect)
{
    iRenTexSizeX = fHUDWidth * fRenderScale;
    iRenTexSizeY = iCustomResY * fRenderScale;
}

if (fAspectRatio < fNativeAspect)
{
    iRenTexSizeX = iCustomResX * fRenderScale;
    iRenTexSizeY = fHUDHeight * fRenderScale;
}
```
The 16:9 aspect ratio of these render textures need to be maintained otherwise you get bugs like this:-

![P3R_2024_02_15_16_34_54_858](https://github.com/Lyall/P3RFix/assets/695941/ed413026-b32a-40f5-81ab-3c7a0b19ecee)

Reply:
Good catch. I don't have an ultrawide so I didn't notice!